kratos/containers/sparse_contiguous_row_graph.h

```diff
@@ -209,58 +209,101 @@ class SparseContiguousRowGraph
         return mGraph;
     }
 
+    //note that this function is slightly less efficient than the span version below
     template<class TVectorType=DenseVector<IndexType>>
     IndexType ExportCSRArrays(
         TVectorType& rRowIndices,
         TVectorType& rColIndices
     ) const
+    {
+        IndexType* pRowIndicesData=nullptr;
+        IndexType RowIndicesDataSize=0;
+        IndexType* pColIndicesData=nullptr;
+        IndexType ColIndicesDataSize=0;
+        ExportCSRArrays(pRowIndicesData,RowIndicesDataSize,pColIndicesData,ColIndicesDataSize);
+        if(rRowIndices.size() != RowIndicesDataSize)
+            rRowIndices.resize(RowIndicesDataSize);
+        IndexPartition<IndexType>(RowIndicesDataSize).for_each( 
+            [&](IndexType i){rRowIndices[i] = pRowIndicesData[i];}
+        );
+        
+        delete [] pRowIndicesData;
+        if(rColIndices.size() != ColIndicesDataSize)
+            rColIndices.resize(ColIndicesDataSize);
+        IndexPartition<IndexType>(ColIndicesDataSize).for_each( 
+            [&](IndexType i){rColIndices[i] = pColIndicesData[i];}
+        );
+        delete [] pColIndicesData;
+
+        return rRowIndices.size();
+    }
+
+    IndexType ExportCSRArrays(
+        span<IndexType>& rRowIndices,
+        span<IndexType>& rColIndices
+    ) const = delete;
+
+    IndexType ExportCSRArrays(
+        IndexType*& pRowIndicesData,
```

Comment:
just to tell that i would have loved to manage the memory by a shared_ptr<TDataType[]> but this is only supported by c++17 or higher.
also i am not completely sure if this would have no performance implications.

actually an even better option would have been to use 
unique_ptr<TDataType[]> but this has difficulties since we may have matrices who do not own the pointer